config/routes.rb

```diff
@@ -1,3 +1,12 @@
 Rails.application.routes.draw do
+  resources :posts
+  resources :comments
+  devise_for :users, :controllers => { registrations: 'registrations' }
+
+  devise_scope :user do  
+    get '/users/sign_out' => 'devise/sessions#destroy'     
+ end
+
+  root 'posts#index'
```

Comment:
The application route should be created with only `resources :posts, only: [:new, :create, :index]`